### PR TITLE
feat: generate/upload sbom for cosign projects

### LINKED ISSUE #959: generating SBOMs for both cosign and cosigned projects and sign them
**Description**

we (w/@dentrax) thought that there are no SBOMs files generated and signed for both `cosign` and `cosigned` projects, so we're proposing to generate, sign them right after generating them via cosign's private key, and attach them to the proper registry. I'd like to list the tools that can help us to do that:

* https://github.com/anchore/syft
* https://github.com/anchore/sbom-action
* https://github.com/CycloneDX/gh-gomod-generate-sbom
* https://github.com/spdx/spdx-sbom-generator
* https://github.com/tern-tools/tern
* https://github.com/CycloneDX/cyclonedx-cli
* https://github.com/kubernetes/release/tree/master/cmd/bom

as an example the `Kyverno` project is already doing this:

👀 github.com/kyverno/kyverno/pull/2438/

cc: @luhring @puerco @cpanato @nishakm

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

#### Summary

feat: generate/upload sbom for cosign projects

#### Ticket Link

Fixes #959 

#### Release Note

```release-note
feat: generate/upload sbom for cosign projects
```

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -34,9 +34,7 @@ jobs:
       - uses: actions/setup-go@v2
         with:
           go-version: '1.17.x'
-      - uses: imjasonh/setup-ko@v0.4
-        with:
-          version: tip
+      - uses: imjasonh/setup-ko@main
```
Comment (imjasonh) [inline .github/workflows/build.yaml:37]: This should still only install the latest release by default. If it doesn't that's a bug.

**Comment by dlorenc** [comment]
Is the vuln stuff supposed to be in here?

**Comment by developer-guy** [comment]
> Is the vuln stuff supposed to be in here?

it is my bad, it should be okay right now.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (developer-guy) [reply .github/workflows/build.yaml:37]: what should I need to do exactly?

Comment (imjasonh) [reply .github/workflows/build.yaml:37]:
```suggestion
      - uses: imjasonh/setup-ko@main
        with:
          version: tip
```

Comment (developer-guy) [reply .github/workflows/build.yaml:37]: ah, yes 🤦

**Comment by developer-guy** [comment]
okay, it seems that it should be ready to merge @dlorenc If I didn't miss anything

**Review by imjasonh — APPROVED** [review]

**Review by dlorenc — APPROVED** [review]

**Pull request merged**